### PR TITLE
Fix digests marshalling

### LINKED ISSUE #18: Fix JSON marshalling for TPM 2.0 digests
This is what it looks like as of now:
https://github.com/orangecms/utk-web/commit/8eb18d0a6262b1d40b17379ca0d62280556d09e9#diff-37dfadeb85117fdfb03addf286bc6002bfac2f93091f7595e1a5b8d39093c199R11

It appears that marshalling heterogeneous data structures in Go is a bit involved. I'm trying to see how it can be done at all.
There is a corresponding `TODO`: https://github.com/9elements/tpmtool/commit/6e6199438513719dad29f5a0198f407663584dcc#diff-be86e8a168afe2b653da62f7d8f9d44c907b55446fb34edb49ff861c01662e7cR143

### PR DESCRIPTION
**Pull request by orangecms**
not elegant, but fixes #18

### CODE REVIEW

File: pkg/tpm/structures.go [hunk 1]
```diff
@@ -126,32 +126,60 @@ type TcgBiosSpecIDEvent struct {
...
-	pcrIndex  uint32 `json:"index"`
+	pcrIndex  uint32
 	eventType uint32
 	digests   LDigestValues
 	eventSize uint32
 	event     []byte
 }
 
-func (e *TcgPcrEvent2) MarshalJSON() ([]byte, error) {
+type MDigest struct {
```
Comment (luxas) [inline pkg/tpm/structures.go:136]:
```go
type mDigest struct {
    Algorithm string `json:"algorithm"
    Digest string `json:"digest"`
}
```

Comment (luxas) [reply pkg/tpm/structures.go:136]: Then you don't need the `MDigest.MarshalJSON`

File: pkg/tpm/structures.go [hunk 2]
```diff
@@ -126,32 +126,60 @@ type TcgBiosSpecIDEvent struct {
...
 	if d != "" {
-		m["data"] = d
+		j, err := json.Marshal(d)
+		if err != nil {
+			return nil, err
+		}
+		m["data"] = json.RawMessage(j)
+	}
+	var ds []MDigest
+	for i := uint32(0); i < e.digests.count; i++ {
```
Comment (luxas) [inline pkg/tpm/structures.go:165]: nit: use `for _, d := range e.digests.digests { }`

Comment (orangecms) [reply pkg/tpm/structures.go]: Thank you!

**Comment by orangecms** [comment]
@zaolin could you recheck the CI integration configuration? It still requires Stickler and Travis to pass, but it's removed from the code already. I cannot access the project settings.

**Pull request closed without merging**